### PR TITLE
cclient: added README.md [skip ci]

### LINKED ISSUE #288: Add CClient README.md

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR fixed #288 

# Test Plan:
No

### CODE REVIEW

File: cclient/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Output  
+
+```
+./bazel-bin/cclient/libapi.a
+./bazel-bin/cclient/libapi.so
+```
+
+## Using CClient library  
+
+C Client library is a subproject under the entangled, for using this library you should create a new bazel project and add this library to the deptendance.
```
Comment (th0br0) [inline cclient/README.md]: `subproject in the entangled repository`

**Comment by th0br0** [comment]
Great! We should provide a CMake-based integration example as well though :thinking:

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: cclient/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    copts = ["-DLOGGER_ENABLE"],
+    srcs = ["cclient_app.c", "cclient_app.h",],
+    includes = ["external/entangled/cclient/"],
+    deps = ["@entangled//cclient:api",],
+    visibility = ["//visibility:public"],
+)
+
+```
+
+C Client application example [iota_cclient_example](https://github.com/oopsmonk/iota_cclient_example)
```
Comment (tsvisabo) [inline cclient/README.md:63]: This README should contain a table listing the API calls too

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/README.md (hunk 1 above)
Comment (jkrvivian) [inline cclient/README.md]: Typo: `deptendance` -> `dependance`

**Comment by oopsmonk** [comment]
> Great! We should provide a CMake-based integration example as well though thinking

For fit this request and support different build systems, we should provide CMakeLists.txt and setup CI for building projects via CMake. In my opinion, we should finish all API implementation first and add CMake to a TODO task.

File: cclient/README.md (hunk 1 above)
Comment (oopsmonk) [reply cclient/README.md]: thanks :-)

File: cclient/README.md (hunk 2 above)
Comment (oopsmonk) [reply cclient/README.md:63]: Sure

File: cclient/README.md [hunk 3]
```diff
@@ -62,3 +62,32 @@ cc_binary(
 ```
 
 C Client application example [iota_cclient_example](https://github.com/oopsmonk/iota_cclient_example)  
+
+## API List  
+C Client API consisted of two API sets:
```
Comment (tsvisabo) [inline cclient/README.md]: consists of

File: cclient/README.md [hunk 4]
```diff
@@ -62,3 +62,32 @@ cc_binary(
 ```
 
 C Client application example [iota_cclient_example](https://github.com/oopsmonk/iota_cclient_example)  
+
+## API List  
+C Client API consisted of two API sets:  
+* **Core APIs** are basic APIs provide by [IRI](https://github.com/iotaledger/iri).
```
Comment (tsvisabo) [inline cclient/README.md]: provided by

**Review by tsvisabo — CHANGES_REQUESTED** [review]

**Review by tsvisabo — APPROVED** [review]

File: cclient/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+* iota_client_get_inclusion_states()
+* iota_client_get_balances()
+* iota_client_get_transactions_to_approve()
+* iota_client_attach_to_tangle()
+* iota_client_interrupt_attaching_to_tangle()
+* iota_client_broadcast_transactions()
+* iota_client_store_transactions()
+* iota_client_check_consistency()
+
+**Extended AIPs**
```
Comment (thibault-martinez) [inline cclient/README.md]: APIs

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/README.md (hunk 5 above)
Comment (oopsmonk) [reply cclient/README.md]: thanks :-)

File: cclient/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+Output  
+
+```
+./bazel-bin/cclient/libapi.a
+./bazel-bin/cclient/libapi.so
+```
+
+## Using CClient library
```
Comment (thibault-martinez) [inline cclient/README.md]: C Client (for consistency)

File: cclient/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+* iota_client_find_transactions()
+* iota_client_get_trytes()
+* iota_client_get_inclusion_states()
+* iota_client_get_balances()
+* iota_client_get_transactions_to_approve()
+* iota_client_attach_to_tangle()
+* iota_client_interrupt_attaching_to_tangle()
+* iota_client_broadcast_transactions()
+* iota_client_store_transactions()
+* iota_client_check_consistency()
```
Comment (thibault-martinez) [inline cclient/README.md:86]: Would be cool to make them link to the API reference.
Or at least provide the link to https://iota.readme.io/v1.5.0/reference somewhere.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Comment by th0br0** [comment]
Better also coordinate with Lewis on the readme.io links - I thought we
were moving those to docs. On a separate platform?

File: cclient/README.md (hunk 7 above)
Comment (oopsmonk) [reply cclient/README.md:86]: The C implementation is slightly different from the online API reference. what I thought is list  APIs here and the user should refer to the Doc which is generated by Doxygen.

**Review by thibault-martinez — APPROVED** [review]

**Review by th0br0 — APPROVED** [review]

**Pull request merged**